mixer/adapter/kubernetesenv/cache.go

```diff
@@ -136,7 +136,8 @@ func (c *controllerImpl) Workload(pod *v1.Pod) workload {
 	wl := workload{name: pod.Name, namespace: pod.Namespace, selfLinkURL: pod.SelfLink}
 	if owner, found := c.rootController(&pod.ObjectMeta); found {
 		wl.name = owner.Name
-		wl.selfLinkURL = fmt.Sprintf("kubernetes://apis/%s/namespaces/%s/%ss/%s", owner.APIVersion, pod.Namespace, strings.ToLower(owner.Kind), owner.Name)
+		wl.selfLinkURL = fmt.Sprintf("kubernetes://apis/%s/namespaces/%s/%ss/%s", owner.APIVersion, pod.Namespace,
+			strcase.CamelCaseToKebabCase(owner.Kind), owner.Name)
```

Comment:
So, this will potentially change something that was:
- `kubernetes://apis/extensions/v1beta1/namespaces/default/replicasets/foo`

into:
- `kubernetes://apis/extensions/v1beta1/namespaces/default/replica-sets/foo`

Is that correct?

I think that is **not** what we want. We need this to match the self-link URLs from the ObjectMetas, iirc.

Ex: `/apis/extensions/v1beta1/namespaces/default/replicasets/sleep-6c66c7765d`

Reply:
Good catch. Reverted.